pkg/daemon/ceph/client/mirror_health.go

```diff
@@ -0,0 +1,225 @@
+		c.interval = &checkInterval.Duration
+	}
+
+	return c
+}
+
+// checkMirroring periodically checks the health of the cluster
+func (c *mirrorChecker) CheckMirroring(context context.Context) {
+	// check the mirroring health immediately before starting the loop
+	err := c.CheckMirroringHealth()
+	if err != nil {
+		c.UpdateStatusMirroring(nil, nil, nil, err.Error())
+		logger.Debugf("failed to check mirroring status for %q. %v", c.namespacedName.Name, err)
+	}
+
+	for {
+		select {
+		case <-context.Done():
+			logger.Infof("stopping monitoring mirroring status for %q", c.namespacedName.Name)
+			return
+
+		case <-time.After(*c.interval):
+			logger.Debugf("checking mirroring status for %q", c.namespacedName.Name)
+			err := c.CheckMirroringHealth()
+			if err != nil {
+				c.UpdateStatusMirroring(nil, nil, nil, err.Error())
+				logger.Debugf("failed to check mirroring status for %q. %v", c.namespacedName.Name, err)
+			}
+		}
+	}
+}
+
+func (c *mirrorChecker) CheckMirroringHealth() error {
+	// Check mirroring status
+	mirrorStatus, err := GetPoolMirroringStatus(c.context, c.clusterInfo, c.monitoringSpec.Name)
+	if err != nil {
+		c.UpdateStatusMirroring(nil, nil, nil, err.Error())
+	}
+
+	// Check mirroring info
+	mirrorInfo, err := GetPoolMirroringInfo(c.context, c.clusterInfo, c.monitoringSpec.Name)
+	if err != nil {
+		c.UpdateStatusMirroring(nil, nil, nil, err.Error())
+	}
+
+	// If snapshot scheduling is enabled let's add it to the status
+	// snapSchedStatus := cephclient.SnapshotScheduleStatus{}
+	snapSchedStatus := []cephv1.SnapshotSchedulesSpec{}
+	if c.monitoringSpec.Mirroring.SnapshotSchedulesEnabled() {
+		snapSchedStatus, err = ListSnapshotSchedulesRecursively(c.context, c.clusterInfo, c.monitoringSpec.Name)
+		if err != nil {
+			c.UpdateStatusMirroring(nil, nil, nil, err.Error())
+		}
+	}
+
+	// On success
+	c.UpdateStatusMirroring(mirrorStatus.Summary, mirrorInfo, snapSchedStatus, "")
+
+	return nil
+}
+
+// updateStatusBucket updates an object with a given status
+func (c *mirrorChecker) UpdateStatusMirroring(mirrorStatus *cephv1.MirroringStatusSummarySpec, mirrorInfo *cephv1.MirroringInfo, snapSchedStatus []cephv1.SnapshotSchedulesSpec, details string) {
+	switch c.objectType.(type) {
+	case *cephv1.CephBlockPool:
+		updatePoolStatusMIrroring(c, mirrorStatus, mirrorInfo, snapSchedStatus, details)
+
+	case *cephv1.CephBlockPoolRadosNamespace:
+		updateRadosNamespaceStatusMirroring(c, mirrorStatus, mirrorInfo, snapSchedStatus, details)
+	}
+}
+
+func updatePoolStatusMIrroring(c *mirrorChecker, mirrorStatus *cephv1.MirroringStatusSummarySpec, mirrorInfo *cephv1.MirroringInfo, snapSchedStatus []cephv1.SnapshotSchedulesSpec, details string) {
+	blockPool := &cephv1.CephBlockPool{}
+	if err := c.client.Get(c.clusterInfo.Context, c.namespacedName, blockPool); err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("CephBlockPool resource not found. Ignoring since object must be deleted.")
```

Comment:
it's helpful to give more details about which object not found, even if just a debug message
```suggestion
			logger.Debug("CephBlockPool %q resource not found for updating the mirroring status, ignoring.", blockPool)
```